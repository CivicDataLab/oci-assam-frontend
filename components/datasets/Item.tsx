/* eslint-disable jsx-a11y/anchor-is-valid */
import Link from 'next/link';
import { Key, ReactChild, ReactFragment, ReactPortal } from 'react';

const Item: React.FC<{ datapackage: any; listType: any }> = ({
  datapackage,
  listType,
}) => {
  return (
    <Link
      href={`/@${
        datapackage.organization ? datapackage.organization.name : 'dataset'
      }/${datapackage.name}`}
    >
      <a className="list__link">
        <p className="list__title">{listType}</p>
        <h3 className="list__heading">
          {datapackage.title || datapackage.name}
        </h3>
        <ul className="list__tags">
          {datapackage.tags
            ?.slice(0, 4)
            .map(
              (tag: {
                id: Key;
                name: boolean | ReactChild | ReactFragment | ReactPortal;
              }) => (
                <li key={tag.id}>{tag.name}</li>
              )
            )}
        </ul>
        {/* <Link
        href={`/@${
          datapackage.organization ? datapackage.organization.name : 'dataset'
        }`}
      >
        <a className="text-gray-500 block mt-1">
          {datapackage.organization
            ? datapackage.organization.title
            : 'dataset'}
        </a>
      </Link> */}
        <p className="list__details">
          {datapackage.description || datapackage.notes}
        </p>
      </a>
    </Link>
  );
};

export default Item;
