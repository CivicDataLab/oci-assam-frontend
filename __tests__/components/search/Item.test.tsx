import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../../../components/search/Item';

test('ðŸ“¸ of Input component with empty', () => {
  const fixture = {
    name: 'qw',
    title: '12',
    organization: null,
    __typename: 'Package',
  };
  const tree = renderer
    .create(<Item datapackage={fixture} key={0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
