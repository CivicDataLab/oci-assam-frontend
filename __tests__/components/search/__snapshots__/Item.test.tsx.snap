// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“¸ of Input component with empty 1`] = `
<div
  className="mb-6 border border-gray-600 p-4"
>
  <h3
    className="text-xl font-semibold"
  >
    <a
      className="text-primary"
      href="/@dataset/qw"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      12
    </a>
  </h3>
  <div
    className="flex items-center flex-wrap mt-4"
  />
  <a
    className="text-gray-500 block mt-1"
    href="/@dataset"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    dataset
  </a>
  <div
    className="leading-relaxed mt-2"
  />
</div>
`;
